### ISSUE TITLE
[LifetimeSafety] False-negative in lifetimebound base class constructor

### ISSUE THREAD

**Issue by usx95** [llvm/llvm-project#192453]
```cpp
struct T {};

struct S {
    S(const T& t [[clang::lifetimebound]]);
};

struct S2 : S {
    S2() : S(T{}) {}  // Here S will outlive T{} but it does not trigger.
};
```
https://godbolt.org/z/WsnWzaEn6
